lib/test-ast.js

```diff
@@ -0,0 +1,91 @@
+'use strict';
+
+const fs = require('fs');
+const AST = require('recast');
+
+/* Lives throughout one test run as `watch` mode is not supported */
+const fileAstCache = {};
+
+/**
+ * Retrieve AST for a test starting at line number with a given title
+ *
+ * @param test      Start line number and title of test function
+ * @param filePath  Path to test file
+ * @return AST      AST of test function
+ */
+module.exports = function (test, filePath) {
+	const {startLineNumber, title} = test || {};
+	if (!startLineNumber) {
+		throw new TypeError('Start line number required.');
+	}
+
+	if (!title) {
+		throw new TypeError('Test title required.');
+	}
+
+	if (!filePath) {
+		throw new TypeError('File path required.');
```

Comment:
Does this really need to be so defensive?

Reply:
Matter of taste, I guess. Personally, I try to assume nothing and be as helpful as possible to the consumer. If TypeScript was used, you could have the best of both worlds, I guess.

Reply:
Hum, interesting view. I guess I don't separate internal and external consumers, but I'll surrender on this one and remove it (and other similar cases).